sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/RowSerializerPoolBase.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.spark.diagnostics.BasicLoggingTrait
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import org.apache.spark.sql.types.StructType
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.{ConcurrentLinkedQueue, Executors, TimeUnit}
+import scala.collection.concurrent.TrieMap
+import scala.util.control.NonFatal
+
+/**
+ * Spark serializers are not thread-safe - and expensive to create (dynamic code generation)
+ * So we will use this object pool to allow reusing serializers based on the targeted schema.
+ * The main purpose for pooling serializers (vs. creating new ones in each PartitionReader) is for Structured
+ * Streaming scenarios where PartitionReaders for the same schema could be created every couple of 100
+ * milliseconds
+ * A clean-up task is used to purge serializers for schemas which weren't used anymore
+ * For each schema we have an object pool that will use a soft-limit to limit the memory footprint
+ */
+abstract class RowSerializerPoolBase[A <: RowSerializerQueueBase] extends BasicLoggingTrait {
```

Comment:
Please make sure this is a rename in Git to allow tracking changes

Reply:
I have reverted the change and then renamed and committed. It was recognized as a renamed by github and git but once I add the new changes it recognizes it as a delete and new file. Not sure what to do. Saw online that git decides based on the percentage of changes.

Reply:
Ah - interesting. Didn't know. Sorry for the extra-work :-(